### PR TITLE
fix bug where the unsaved changes modal was running even when nothing…

### PR DESCRIPTION
**Pull request by toastynerd**
@joshreeves Some changes to more accurately tell when the content has actually been modified.

### CODE REVIEW

File: vendor/assets/javascripts/behaviors/nested_resources.js [hunk 1]
```diff
@@ -135,9 +135,15 @@ Cmsify.NestedResource.prototype.deserializeFields = function(fields) {
...
 
 Cmsify.NestedResource.prototype.setIsModified = function () {
-  if (tinyMCE.editors.some(function(editor) { return editor.isDirty() }) ||
-      JSON.stringify(this.serializeFields()) != JSON.stringify(this.lastData)) {
+  var tinyMCEDirty = false;
+  tinyMCE.editors.some(function(editor) {
+    if (editor.isDirty()) tinyMCEDirty = true;
+  }.bind(this));
+
+  if (JSON.stringify(this.serializeFields()).localeCompare(JSON.stringify(this.lastData)) != 0) {
```
Comment (joshreeves) [inline vendor/assets/javascripts/behaviors/nested_resources.js]: Why does this fix it?

Comment (toastynerd) [reply vendor/assets/javascripts/behaviors/nested_resources.js]: It actually doesn't, it's the other part with parsing the `isDirty()` it was just the first thing I tried and I didn't change it back. I'll change it back.

Comment (joshreeves) [inline vendor/assets/javascripts/behaviors/nested_resources.js]: @toastynerd - I don't see how any of this tinyMCE functionality is changing. Can you explain exactly what this PR is changing and why it fixes the issue?

File: vendor/assets/javascripts/behaviors/nested_resources.js [hunk 2]
```diff
@@ -135,9 +135,15 @@ Cmsify.NestedResource.prototype.deserializeFields = function(fields) {
...
-      JSON.stringify(this.serializeFields()) != JSON.stringify(this.lastData)) {
+  var tinyMCEDirty = false;
+  tinyMCE.editors.some(function(editor) {
+    if (editor.isDirty()) tinyMCEDirty = true;
+  }.bind(this));
+
+  if (JSON.stringify(this.serializeFields()) !== JSON.stringify(this.lastData)) {
     this.isModified = true;
+  } else {
+    this.isModified = tinyMCEDirty;
```
Comment (joshreeves) [inline vendor/assets/javascripts/behaviors/nested_resources.js]: @toastynerd - I don't see how any of this tinyMCE functionality is changing. Can you explain exactly what this PR is changing and why it fixes the issue?

Comment (toastynerd) [reply vendor/assets/javascripts/behaviors/nested_resources.js]: Basically we were setting isModified to true and there was no way to set it back to false if the form was reverted to it's original state, I can actually probably do it with the old style, let me give it a shot.

**Comment by toastynerd** [comment]
@joshreeves Okay, now this should be a little more clear. We'll still get a false unsaved changes modal when editing the tinyMCE boxes because of the way they do dirty checking. They consider anything that creates an edit history as "dirty" instead of basing it on the actual content but it should work in all the other places of the app.

**Comment by joshreeves** [comment]
🚢

**Pull request merged**